src/main/java/com/google/research/bleth/servlets/DeleteSimulationServlet.java

```diff
@@ -0,0 +1,27 @@
+package com.google.research.bleth.servlets;
+
+import com.google.research.bleth.utils.Queries;
+
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** A servlet used for deleting simulations. */
+@WebServlet("/delete-simulation")
+public class DeleteSimulationServlet extends HttpServlet {
+
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String simulationId = request.getParameter("simulationId");
+        String responseText = "Simulation has been deleted successfully.";
+        try {
+            Queries.delete(simulationId);
+        } catch (Exception e) {
+            responseText = "Something went wrong: " + e.getMessage();
+        }
+        response.setContentType("text/html;");
+        response.getWriter().println(responseText);
```

Comment:
This is generating invalid HTML - (1) you're missing (at minimum) <html><body> ... </body></html>, (2) if the exception message includes any special characters they won't be escaped (and can confuse the parsers even further).

Either change the response type to text/plain, or properly wrap and HTML-escape your output.